src/main/scala/is/hail/keytable/KeyTable.scala

```diff
@@ -484,6 +489,72 @@ case class KeyTable(hc: HailContext, rdd: RDD[Row],
     KeyTable(hc, joinedRDD, newSignature, key)
   }
 
+  def broadcastLeftJoinDistinct(right: KeyTable): KeyTable = {
+    if (key.length != right.key.length || !(keyFields.map(_.typ) sameElements right.keyFields.map(_.typ)))
+      fatal(
+        s"""Both key tables must have the same number of keys and the types of keys must be identical. Order matters.
+           |  Left signature: ${ TStruct(keyFields).toPrettyString(compact = true) }
+           |  Right signature: ${ TStruct(right.keyFields).toPrettyString(compact = true) }""".stripMargin)
+
+    val overlappingFields = fieldNames.toSet.intersect(right.fieldNames.toSet) -- key -- right.key
+    if (overlappingFields.nonEmpty)
+      fatal(
+        s"""Columns that are not keys cannot be present in both key tables.
+           |  Overlapping fields: ${ overlappingFields.mkString(", ") }""".stripMargin)
+
+    val newSignature = TStruct((keySignature.fields ++ valueSignature.fields ++ right.valueSignature.fields)
+      .map(fd => (fd.name, fd.typ)): _*)
+    val localNKeys = nKeys
+    val size1 = valueSignature.size
+    val size2 = right.valueSignature.size
+    val totalSize = newSignature.size
+
+    assert(totalSize == localNKeys + size1 + size2)
+
+    val merger = (k: Row, r1: Row, r2: Row) => {
```

Comment:
The code copied from `join` makes me feel even more strongly that the two methods should be unified in `join`.